### ISSUE TITLE
Error decompiling method com.codename1.components.ToastBar.updateStatus()V part 5

### ISSUE THREAD

**Issue by shannah** [konsoletyper/teavm#316]
I get the following error when I try to build a project with the latest TeaVM 0.6.0-snapshot.  The same error occurs (except on a different class) with 0.5.0.

Project is https://github.com/shannah/teavm-060-error-decompiling-method

ToastBar class source seen [here](https://github.com/codenameone/CodenameOne/blob/master/CodenameOne/src/com/codename1/components/ToastBar.java)

~~~~
[java] [ERROR] Failed to execute goal org.teavm:teavm-maven-plugin:0.6.0-SNAPSHOT:compile (web-client) on project KitchenSink: Unexpected error occured: Error decompiling method com.codename1.components.ToastBar.updateStatus()V part 5:
     [java] [ERROR] var @this as this
     [java] [ERROR] var @c as c
     [java] [ERROR] var @5 as s
     [java] [ERROR] var @6 as s
     [java] [ERROR] var @7 as s
     [java] [ERROR] var @13 as c
     [java] [ERROR] var @14 as s
     [java] [ERROR] var @15 as s
     [java] [ERROR] var @16 as oldText
     [java] [ERROR] var @20 as s
     [java] [ERROR] var @21 as s
     [java] [ERROR] var @22 as s
     [java] [ERROR] var @s as s
     [java] [ERROR] var @l as l
     [java] [ERROR] var @26 as oldText
     [java] [ERROR] var @27 as oldText
     [java] [ERROR] var @28 as oldText
     [java] [ERROR] var @29 as oldText
     [java] [ERROR] var @30 as oldText
     [java] [ERROR] var @31 as newLabel
     [java] [ERROR] var @32 as oldText
     [java] [ERROR] var @33 as newLabel
     [java] [ERROR] var @34 as oldText
     [java] [ERROR] var @36 as oldText
     [java] [ERROR] var @38 as oldText
     [java] [ERROR] var @40 as oldText
     [java] [ERROR] var @41 as oldText
     [java] [ERROR] var @43 as oldText
     [java] [ERROR] var @44 as anim
     [java] [ERROR] var @anim as oldText
     [java] [ERROR] var @49 as oldText
     [java] [ERROR] var @51 as oldText
     [java] [ERROR] var @52 as newLabel
     [java] [ERROR] var @53 as oldText
     [java] [ERROR] var @61 as oldText
     [java] [ERROR] var @62 as oldText
     [java] [ERROR] var @65 as oldText
     [java] [ERROR] var @67 as oldText
     [java] [ERROR] var @69 as oldText
     [java] [ERROR] var @71 as oldText
     [java] [ERROR] var @72 as newLabel
     [java] [ERROR] var @73 as oldText
     [java] [ERROR] var @74 as newLabel
     [java] [ERROR] var @76 as oldText
     [java] [ERROR] var @77 as oldText
     [java] [ERROR] var @78 as oldText
     [java] [ERROR] var @79 as oldText
     [java] [ERROR] var @81 as oldText
     [java] [ERROR] var @82 as newLabel
     [java] [ERROR] var @83 as oldText
     [java] [ERROR] var @85 as oldText
     [java] [ERROR] var @86 as oldText
     [java] [ERROR] var @87 as oldText
     [java] [ERROR] var @88 as newLabel
     [java] [ERROR] var @90 as oldText
     [java] [ERROR] var @91 as oldText
     [java] [ERROR] var @92 as oldText
     [java] [ERROR] var @oldText as oldText
     [java] [ERROR] var @95 as oldText
     [java] [ERROR] var @96 as oldText
     [java] [ERROR] var @97 as oldText
     [java] [ERROR] var @98 as newLabel
     [java] [ERROR] var @99 as oldText
     [java] [ERROR] var @100 as newLabel
     [java] [ERROR] var @101 as oldText
     [java] [ERROR] var @102 as s
     [java] [ERROR] var @103 as s
     [java] [ERROR] var @104 as s
     [java] [ERROR] var @105 as s
     [java] [ERROR] var @106 as oldText
     [java] [ERROR] var @107 as s
     [java] [ERROR] var @109 as newLabel
     [java] [ERROR] var @112 as oldText
     [java] [ERROR] var @113 as oldText
     [java] [ERROR] var @114 as oldText
     [java] [ERROR] var @newLabel as oldText
     [java] [ERROR] var @116 as oldText
     [java] [ERROR] var @117 as oldText
     [java] [ERROR] var @118 as oldText
     [java] [ERROR] var @119 as s
     [java] [ERROR] var @120 as s
     [java] [ERROR] var @121 as s
     [java] [ERROR] var @122 as s
     [java] [ERROR] var @124 as s
     [java] [ERROR] var @125 as s
     [java] [ERROR] var @126 as s
     [java] [ERROR] var @127 as oldText
     [java] [ERROR] var @129 as oldTextAreaSize
     [java] [ERROR] var @130 as s
     [java] [ERROR] var @131 as s
     [java] [ERROR] var @133 as newTexAreaSize
     [java] [ERROR] var @134 as s
     [java] [ERROR] var @oldTextAreaSize as s
     [java] [ERROR] var @136 as s
     [java] [ERROR] var @137 as s
     [java] [ERROR] var @138 as oldText
     [java] [ERROR] var @141 as l
     [java] [ERROR] var @145 as s
     [java] [ERROR] var @147 as c
     [java] [ERROR] var @152 as c
     [java] [ERROR] var @153 as s
     [java] [ERROR] var @154 as oldText
     [java] [ERROR] var @155 as oldText
     [java] [ERROR] var @156 as newLabel
     [java] [ERROR] var @157 as oldText
     [java] [ERROR] var @158 as newLabel
     [java] [ERROR] var @159 as s
     [java] [ERROR] var @160 as oldText
     [java] [ERROR] var @162 as s
     [java] [ERROR] var @167 as c
     [java] [ERROR] var @168 as s
     [java] [ERROR] var @169 as c
     [java] [ERROR] var @173 as s
     [java] [ERROR] var @174 as oldText
     [java] [ERROR] $0
     [java] [ERROR] goto $90
     [java] [ERROR] catch  goto $88
     [java] [ERROR] $1
     [java] [ERROR] $2
     [java] [ERROR] $3
     [java] [ERROR] $4
     [java] [ERROR] $5
     [java] [ERROR] $6
     [java] [ERROR] $7
     [java] [ERROR] $8
     [java] [ERROR] $9
     [java] [ERROR] $10
     [java] [ERROR] $11
     [java] [ERROR] at 'com/codename1/components/ToastBar.java' 499
     [java] [ERROR] @7 := field com.codename1.components.ToastBar.statuses @this as `Ljava/util/ArrayList;`
     [java] [ERROR] catch  goto $88
     [java] [ERROR] $12
     [java] [ERROR] $13
     [java] [ERROR] at 'com/codename1/components/ToastBar.java' 507
     [java] [ERROR] @s := invokeStatic `com.codename1.components.ToastBar$ToastBarComponent.access$500(Lcom/codename1/components/ToastBar$ToastBarComponent;)Lcom/codename1/components/ToastBar$Status;` @c
     [java] [ERROR] at 'com/codename1/components/ToastBar.java' 509
     [java] [ERROR] nop
     [java] [ERROR] @l := new com.codename1.ui.Label
     [java] [ERROR] @26 := invoke `com.codename1.components.ToastBar$Status.getMessage()Ljava/lang/String;` @s
     [java] [ERROR] if @26 === null then goto $16 else goto $17
     [java] [ERROR] catch  goto $88
     [java] [ERROR] $14
     [java] [ERROR] $15
     [java] [ERROR] $16
     [java] [ERROR] at 'com/codename1/components/ToastBar.java' 509
     [java] [ERROR] @27 := ''
     [java] [ERROR] at unknown location
     [java] [ERROR] goto $18
     [java] [ERROR] catch  goto $88
     [java] [ERROR] $17
     [java] [ERROR] at 'com/codename1/components/ToastBar.java' 509
     [java] [ERROR] @27 := invoke `com.codename1.components.ToastBar$Status.getMessage()Ljava/lang/String;` @s
     [java] [ERROR] goto $18
     [java] [ERROR] catch  goto $88
     [java] [ERROR] $18
     [java] [ERROR] catch  goto $88
     [java] [ERROR] catch  goto $88
     [java] [ERROR] $19
     [java] [ERROR] $20
     [java] [ERROR] $21
     [java] [ERROR] $22
     [java] [ERROR] $23
     [java] [ERROR] $24
     [java] [ERROR] $25
     [java] [ERROR] $26
     [java] [ERROR] $27
     [java] [ERROR] $28
     [java] [ERROR] $29
     [java] [ERROR] $30
     [java] [ERROR] $31
     [java] [ERROR] $32
     [java] [ERROR] $33
     [java] [ERROR] $34
     [java] [ERROR] $35
     [java] [ERROR] $36
     [java] [ERROR] $37
     [java] [ERROR] $38
     [java] [ERROR] $39
     [java] [ERROR] $40
     [java] [ERROR] $41
     [java] [ERROR] $42
     [java] [ERROR] $43
     [java] [ERROR] $44
     [java] [ERROR] $45
     [java] [ERROR] $46
     [java] [ERROR] $47
     [java] [ERROR] $48
     [java] [ERROR] $49
     [java] [ERROR] $50
     [java] [ERROR] $51
     [java] [ERROR] $52
     [java] [ERROR] $53
     [java] [ERROR] $54
     [java] [ERROR] $55
     [java] [ERROR] $56
     [java] [ERROR] $57
     [java] [ERROR] $58
     [java] [ERROR] $59
     [java] [ERROR] $60
     [java] [ERROR] $61
     [java] [ERROR] $62
     [java] [ERROR] $63
     [java] [ERROR] $64
     [java] [ERROR] $65
     [java] [ERROR] $66
     [java] [ERROR] $67
     [java] [ERROR] $68
     [java] [ERROR] $69
     [java] [ERROR] $70
     [java] [ERROR] $71
     [java] [ERROR] $72
     [java] [ERROR] $73
     [java] [ERROR] $74
     [java] [ERROR] $75
     [java] [ERROR] $76
     [java] [ERROR] $77
     [java] [ERROR] $78
     [java] [ERROR] $79
     [java] [ERROR] $80
     [java] [ERROR] $81
     [java] [ERROR] $82
     [java] [ERROR] $83
     [java] [ERROR] $84
     [java] [ERROR] $85
     [java] [ERROR] at 'com/codename1/components/ToastBar.java' 640
     [java] [ERROR] throw @169
     [java] [ERROR] $86
     [java] [ERROR] at 'com/codename1/components/ToastBar.java' 639
     [java] [ERROR] @172 := 0
     [java] [ERROR] field com.codename1.components.ToastBar.pendingUpdateStatus @this := @172 as Z
     [java] [ERROR] at 'com/codename1/components/ToastBar.java' 640
     [java] [ERROR] @173 := invokeStatic `com.codename1.ui.Display.getInstance()Lcom/codename1/ui/Display;`
     [java] [ERROR] @174 := new com.codename1.components.ToastBar$2
     [java] [ERROR] invoke `com.codename1.components.ToastBar$2.<init>(Lcom/codename1/components/ToastBar;)V` @174, @this
     [java] [ERROR] $87
     [java] [ERROR] $88
     [java] [ERROR] @169 := exception
     [java] [ERROR] at 'com/codename1/components/ToastBar.java' 637
     [java] [ERROR] @170 := 0
     [java] [ERROR] field com.codename1.components.ToastBar.updatingStatus @this := @170 as Z
     [java] [ERROR] at 'com/codename1/components/ToastBar.java' 638
     [java] [ERROR] @171 := field com.codename1.components.ToastBar.pendingUpdateStatus @this as Z
     [java] [ERROR] if @171 == 0 then goto $85 else goto $86
     [java] [ERROR] $89
     [java] [ERROR] $90
     [java] [ERROR] at 'com/codename1/components/ToastBar.java' 498
     [java] [ERROR] @23 := invoke `java.util.AbstractCollection.isEmpty()Z` @22
     [java] [ERROR] if @23 == 0 then goto $13 else goto $11
     [java] [ERROR] catch  goto $88: fromIndex(2) > toIndex(1)
     [java] [ERROR] -> [Help 1]
     [java] org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.teavm:teavm-maven-plugin:0.6.0-SNAPSHOT:compile (web-client) on project KitchenSink: Unexpected error occured
     [java] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:216)
     [java] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
     [java] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
     [java] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
     [java] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
     [java] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
     [java] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
     [java] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:347)
     [java] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:154)
     [java] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:582)
     [java] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:214)
     [java] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:158)
     [java] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [java] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     [java] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     [java] 	at java.lang.reflect.Method.invoke(Method.java:483)
     [java] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
     [java] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
     [java] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
     [java] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
     [java] Caused by: org.apache.maven.plugin.MojoExecutionException: Unexpected error occured
     [java] 	at org.teavm.maven.TeaVMCompileMojo.execute(TeaVMCompileMojo.java:111)
     [java] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
     [java] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
     [java] 	... 19 more
     [java] Caused by: org.teavm.ast.decompilation.DecompilationException: Error decompiling method com.codename1.components.ToastBar.updateStatus()V part 5:
     [java]   var @this as this
     [java]   var @c as c
     [java]   var @5 as s
     [java]   var @6 as s
     [java]   var @7 as s
     [java]   var @13 as c
     [java]   var @14 as s
     [java]   var @15 as s
     [java]   var @16 as oldText
     [java]   var @20 as s
     [java]   var @21 as s
     [java]   var @22 as s
     [java]   var @s as s
     [java]   var @l as l
     [java]   var @26 as oldText
     [java]   var @27 as oldText
     [java]   var @28 as oldText
     [java]   var @29 as oldText
     [java]   var @30 as oldText
     [java]   var @31 as newLabel
     [java]   var @32 as oldText
     [java]   var @33 as newLabel
     [java]   var @34 as oldText
     [java]   var @36 as oldText
     [java]   var @38 as oldText
     [java]   var @40 as oldText
     [java]   var @41 as oldText
     [java]   var @43 as oldText
     [java]   var @44 as anim
     [java]   var @anim as oldText
     [java]   var @49 as oldText
     [java]   var @51 as oldText
     [java]   var @52 as newLabel
     [java]   var @53 as oldText
     [java]   var @61 as oldText
     [java]   var @62 as oldText
     [java]   var @65 as oldText
     [java]   var @67 as oldText
     [java]   var @69 as oldText
     [java]   var @71 as oldText
     [java]   var @72 as newLabel
     [java]   var @73 as oldText
     [java]   var @74 as newLabel
     [java]   var @76 as oldText
     [java]   var @77 as oldText
     [java]   var @78 as oldText
     [java]   var @79 as oldText
     [java]   var @81 as oldText
     [java]   var @82 as newLabel
     [java]   var @83 as oldText
     [java]   var @85 as oldText
     [java]   var @86 as oldText
     [java]   var @87 as oldText
     [java]   var @88 as newLabel
     [java]   var @90 as oldText
     [java]   var @91 as oldText
     [java]   var @92 as oldText
     [java]   var @oldText as oldText
     [java]   var @95 as oldText
     [java]   var @96 as oldText
     [java]   var @97 as oldText
     [java]   var @98 as newLabel
     [java]   var @99 as oldText
     [java]   var @100 as newLabel
     [java]   var @101 as oldText
     [java]   var @102 as s
     [java]   var @103 as s
     [java]   var @104 as s
     [java]   var @105 as s
     [java]   var @106 as oldText
     [java]   var @107 as s
     [java]   var @109 as newLabel
     [java]   var @112 as oldText
     [java]   var @113 as oldText
     [java]   var @114 as oldText
     [java]   var @newLabel as oldText
     [java]   var @116 as oldText
     [java]   var @117 as oldText
     [java]   var @118 as oldText
     [java]   var @119 as s
     [java]   var @120 as s
     [java]   var @121 as s
     [java]   var @122 as s
     [java]   var @124 as s
     [java]   var @125 as s
     [java]   var @126 as s
     [java]   var @127 as oldText
     [java]   var @129 as oldTextAreaSize
     [java]   var @130 as s
     [java]   var @131 as s
     [java]   var @133 as newTexAreaSize
     [java]   var @134 as s
     [java]   var @oldTextAreaSize as s
     [java]   var @136 as s
     [java]   var @137 as s
     [java]   var @138 as oldText
     [java]   var @141 as l
     [java]   var @145 as s
     [java]   var @147 as c
     [java]   var @152 as c
     [java]   var @153 as s
     [java]   var @154 as oldText
     [java]   var @155 as oldText
     [java]   var @156 as newLabel
     [java]   var @157 as oldText
     [java]   var @158 as newLabel
     [java]   var @159 as s
     [java]   var @160 as oldText
     [java]   var @162 as s
     [java]   var @167 as c
     [java]   var @168 as s
     [java]   var @169 as c
     [java]   var @173 as s
     [java]   var @174 as oldText
     [java]   $0
     [java]       goto $90
     [java]       catch  goto $88
     [java]   $1
     [java]   $2
     [java]   $3
     [java]   $4
     [java]   $5
     [java]   $6
     [java]   $7
     [java]   $8
     [java]   $9
     [java]   $10
     [java]   $11
     [java]     at 'com/codename1/components/ToastBar.java' 499
     [java]       @7 := field com.codename1.components.ToastBar.statuses @this as `Ljava/util/ArrayList;`
     [java]       catch  goto $88
     [java]   $12
     [java]   $13
     [java]     at 'com/codename1/components/ToastBar.java' 507
     [java]       @s := invokeStatic `com.codename1.components.ToastBar$ToastBarComponent.access$500(Lcom/codename1/components/ToastBar$ToastBarComponent;)Lcom/codename1/components/ToastBar$Status;` @c
     [java]     at 'com/codename1/components/ToastBar.java' 509
     [java]       nop
     [java]       @l := new com.codename1.ui.Label
     [java]       @26 := invoke `com.codename1.components.ToastBar$Status.getMessage()Ljava/lang/String;` @s
     [java]       if @26 === null then goto $16 else goto $17
     [java]       catch  goto $88
     [java]   $14
     [java]   $15
     [java]   $16
     [java]     at 'com/codename1/components/ToastBar.java' 509
     [java]       @27 := ''
     [java]     at unknown location
     [java]       goto $18
     [java]       catch  goto $88
     [java]   $17
     [java]     at 'com/codename1/components/ToastBar.java' 509
     [java]       @27 := invoke `com.codename1.components.ToastBar$Status.getMessage()Ljava/lang/String;` @s
     [java]       goto $18
     [java]       catch  goto $88
     [java]   $18
     [java]       catch  goto $88
     [java]       catch  goto $88
     [java]   $19
     [java]   $20
     [java]   $21
     [java]   $22
     [java]   $23
     [java]   $24
     [java]   $25
     [java]   $26
     [java]   $27
     [java]   $28
     [java]   $29
     [java]   $30
     [java]   $31
     [java]   $32
     [java]   $33
     [java]   $34
     [java]   $35
     [java]   $36
     [java]   $37
     [java]   $38
     [java]   $39
     [java]   $40
     [java]   $41
     [java]   $42
     [java]   $43
     [java]   $44
     [java]   $45
     [java]   $46
     [java]   $47
     [java]   $48
     [java]   $49
     [java]   $50
     [java]   $51
     [java]   $52
     [java]   $53
     [java]   $54
     [java]   $55
     [java]   $56
     [java]   $57
     [java]   $58
     [java]   $59
     [java]   $60
     [java]   $61
     [java]   $62
     [java]   $63
     [java]   $64
     [java]   $65
     [java]   $66
     [java]   $67
     [java]   $68
     [java]   $69
     [java]   $70
     [java]   $71
     [java]   $72
     [java]   $73
     [java]   $74
     [java]   $75
     [java]   $76
     [java]   $77
     [java]   $78
     [java]   $79
     [java]   $80
     [java]   $81
     [java]   $82
     [java]   $83
     [java]   $84
     [java]   $85
     [java]     at 'com/codename1/components/ToastBar.java' 640
     [java]       throw @169
     [java]   $86
     [java]     at 'com/codename1/components/ToastBar.java' 639
     [java]       @172 := 0
     [java]       field com.codename1.components.ToastBar.pendingUpdateStatus @this := @172 as Z
     [java]     at 'com/codename1/components/ToastBar.java' 640
     [java]       @173 := invokeStatic `com.codename1.ui.Display.getInstance()Lcom/codename1/ui/Display;`
     [java]       @174 := new com.codename1.components.ToastBar$2
     [java]       invoke `com.codename1.components.ToastBar$2.<init>(Lcom/codename1/components/ToastBar;)V` @174, @this
     [java]   $87
     [java]   $88
     [java]     @169 := exception
     [java]     at 'com/codename1/components/ToastBar.java' 637
     [java]       @170 := 0
     [java]       field com.codename1.components.ToastBar.updatingStatus @this := @170 as Z
     [java]     at 'com/codename1/components/ToastBar.java' 638
     [java]       @171 := field com.codename1.components.ToastBar.pendingUpdateStatus @this as Z
     [java]       if @171 == 0 then goto $85 else goto $86
     [java]   $89
     [java]   $90
     [java]     at 'com/codename1/components/ToastBar.java' 498
     [java]       @23 := invoke `java.util.AbstractCollection.isEmpty()Z` @22
     [java]       if @23 == 0 then goto $13 else goto $11
     [java]       catch  goto $88
     [java] 
     [java] 	at org.teavm.ast.decompilation.Decompiler.decompileAsyncCacheMiss(Decompiler.java:336)
     [java] 	at org.teavm.ast.decompilation.Decompiler.decompileAsync(Decompiler.java:295)
     [java] 	at org.teavm.ast.decompilation.Decompiler.decompile(Decompiler.java:219)
     [java] 	at org.teavm.ast.decompilation.Decompiler.decompile(Decompiler.java:208)
     [java] 	at org.teavm.backend.javascript.JavaScriptTarget.modelToAst(JavaScriptTarget.java:332)
     [java] 	at org.teavm.backend.javascript.JavaScriptTarget.emit(JavaScriptTarget.java:238)
     [java] 	at org.teavm.backend.javascript.JavaScriptTarget.emit(JavaScriptTarget.java:226)
     [java] 	at org.teavm.vm.TeaVM.build(TeaVM.java:397)
     [java] 	at org.teavm.tooling.TeaVMTool.generate(TeaVMTool.java:423)
     [java] 	at org.teavm.maven.TeaVMCompileMojo.execute(TeaVMCompileMojo.java:106)
     [java] 	... 21 more
     [java] Caused by: java.lang.IllegalArgumentException: fromIndex(2) > toIndex(1)
     [java] 	at java.util.ArrayList.subListRangeCheck(ArrayList.java:991)
     [java] 	at java.util.ArrayList.subList(ArrayList.java:981)
     [java] 	at org.teavm.ast.decompilation.Decompiler.closeExpiredBookmarks(Decompiler.java:532)
     [java] 	at org.teavm.ast.decompilation.Decompiler.getRegularMethodStatement(Decompiler.java:403)
     [java] 	at org.teavm.ast.decompilation.Decompiler.decompileAsyncCacheMiss(Decompiler.java:331)
     [java] 	... 30 more
     [java] [ERROR] 
     [java] [ERROR] Re-run Maven using the -X switch to enable full debug logging.
     [java] [ERROR] 
     [java] [ERROR] For more information about the errors and possible solutions, please read the following articles:
     [java] [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
~~~~

**Comment by shooshx** [comment]
Where did that project go?
I'm seeing the same error in a code snippet that has a break from a while loop

**Comment by konsoletyper** [comment]
@shooshx what do you mean?

Can you please provide a project that reproduces your issue?

**Comment by shooshx** [comment]
shannah above posted a link to a repository that shows the error
https://github.com/shannah/teavm-060-error-decompiling-method
but it's a dead link right now.
I'll try to isolate that bit of code, it's part of a rather big project

**Comment by konsoletyper** [comment]
I see. It's a private repository, Steve gave me access to the repository, so I can clone it and reproduce the issue.

**Comment by konsoletyper** [comment]
Fixed by 0c8013dfcf3344389e9f209904442057b60b9af5
@shannah please, verify

**Comment by shannah** [comment]
Thanks.  Missed this when you pinged me the first time.  I just verified that it fixes the issue.
